pkg/kubernetes/scripts/virtual-machines/components/VmsListing.jsx

```diff
@@ -26,10 +26,13 @@ import { gettext as _ } from 'cockpit'
 import { Listing } from '../../../../lib/cockpit-components-listing.jsx'
 import VmsListingRow from './VmsListingRow.jsx'
 
-const VmsListing = ({ vms, settings, vmsMessages }) => {
+const VmsListing = ({ vms, settings, vmsMessages, pods }) => {
     const isOpenshift = settings.flavor === 'openshift'
     const namespaceLabel = isOpenshift ? _("Project") : _("Namespace")
-    const rows = vms.map(vm => (<VmsListingRow vm={vm} vmMessages={vmsMessages[vm.metadata.uid]} key={vm.metadata.uid} />))
+    const rows = vms.map(vm => (<VmsListingRow vm={vm}
+                                               vmMessages={vmsMessages[vm.metadata.uid]}
+                                               pods={pods}
```

Comment:
Suggestion: filter the correct pod from the list here already, and only hand over the actual pod that belongs to `vm`, instead of the whole list. Structurally it seems to me that components that show a single VM don't need to make assumptions and have knowledge about the entirety of pods. But fine for me if you prefer this.